### PR TITLE
[SE-0274] Enable concise #file by default

### PR DESCRIPTION
**Pull request by beccadax**
This pull request completes the implementation of [SE-0274 "Concise magic file names"](https://github.com/apple/swift-evolution/blob/master/proposals/0274-magic-file.md):

* The new `#file` strings are now enabled by default.
* The `-enable-concise-pound-file` flag forcibly enables the feature; the `-disable-concise-pound-file` flag disables it. The old `-enable-experimental-concise-pound-file` flag remains as an alias for `-enable-concise-pound-file`.
* Enabling `#file` by default revealed that a couple of tests which call the frontend directly have been violating the same-name rule. I've modified those tests and made the frontend handle this more gracefully.
* Corrects a small `#filePath` code completion bug.

Fixes <rdar://56816166>.

### CODE REVIEW

File: test/IDE/complete_pound_expr.swift [hunk 1]
```diff
@@ -15,26 +15,29 @@ func test1() {
   let _ = useSelector(##^POUND_EXPR_3^#)
 }
 
-// POUND_EXPR_INTCONTEXT: Begin completions, 5 items
+// POUND_EXPR_INTCONTEXT: Begin completions, 6 items
 // POUND_EXPR_INTCONTEXT-DAG: Keyword[#function]/None:            function[#String#]; name=function
 // POUND_EXPR_INTCONTEXT-DAG: Keyword[#file]/None:                file[#String#]; name=file
+// POUND_EXPR_INTCONTEXT-DAG: Keyword[#file]/None: filePath[#String#]; name=filePath
```
Comment (brentdax) [inline test/IDE/complete_pound_expr.swift:21]: @rintaro I don't really know what all of the fields in these lines mean. Is it correct to have `Keyword[#file]/None: filePath`, or does something need to be added to code completion so we get `Keyword[#filePath]` instead?

Comment (rintaro) [reply test/IDE/complete_pound_expr.swift:21]: We want `#filePath` here. You can make it by changing https://github.com/brentdax/swift/blob/0f0622d22b6d8e3327acc0ecff8967272342b838/lib/IDE/CodeCompletion.cpp#L3878
to `CodeCompletionKeywordKind::pound_filePath`.

**Comment by brentdax** [comment]
@swift-ci please test

**Comment by brentdax** [comment]
In discussions with standard library contributors, they preferred for StdlibUnittest and the other standard library testing helpers to continue using full paths, so I've switched them to `#filePath`.

**Comment by brentdax** [comment]
@swift-ci please test

**Comment by brentdax** [comment]
SwiftPM has some uses of `#file` to locate fixtures.

**Comment by brentdax** [comment]
With apple/swift-package-manager#2688

@swift-ci please test

**Pull request closed without merging**